File: app/src/main/java/com/kyhsgeekcode/disassembler/AssemblyProvider.java
package com.kyhsgeekcode.disassembler;

import android.os.Handler;
import android.os.Looper;

import java.util.List;

public abstract class AssemblyProvider {
    public abstract long getAll(int handle, byte[] bytes, long offset, long size, long virtaddr);

    public abstract long getSome(int handle, byte[] bytes, long offset, long size, long virtaddr, int count);
}

File: app/src/main/java/com/kyhsgeekcode/disassembler/AssemblyProvider.kt
package com.kyhsgeekcode.disassembler

abstract class AssemblyProvider {
    abstract fun getAll(bytes: ByteArray, offset: Long, size: Long, virtaddr: Long): Long
    abstract fun getSome(
        bytes: ByteArray,
        offset: Long,
        size: Long,
        virtaddr: Long,
        count: Int
    ): List<DisassemblyListItem>
}